electrum/gui/qt/wizard/wallet.py

```diff
@@ -0,0 +1,1322 @@
+
+        path = wizard._path
+
+        if os.path.isdir(path):
+            raise Exception("wallet path cannot point to a directory")
+
+        self.wallet_exists = False
+        self.wallet_is_open = False
+        self.wallet_needs_hw_unlock = False
+
+        hbox = QHBoxLayout()
+        hbox.addWidget(QLabel(_('Wallet') + ':'))
+        self.name_e = QLineEdit()
+        hbox.addWidget(self.name_e)
+        button = QPushButton(_('Choose...'))
+        hbox.addWidget(button)
+        self.layout().addLayout(hbox)
+
+        msg_label = WWLabel('')
+        self.layout().addWidget(msg_label)
+        hbox2 = QHBoxLayout()
+        self.pw_e = PasswordLineEdit('', self)
+        self.pw_e.setFixedWidth(17 * char_width_in_lineedit())
+        pw_label = QLabel(_('Password') + ':')
+        hbox2.addWidget(pw_label)
+        hbox2.addWidget(self.pw_e)
+        hbox2.addStretch()
+        self.layout().addLayout(hbox2)
+
+        self.layout().addSpacing(50)
+        vbox_create_new = QVBoxLayout()
+        vbox_create_new.addWidget(QLabel(_('Alternatively') + ':'), alignment=Qt.AlignLeft)
+        button_create_new = QPushButton(_('Create New Wallet'))
+        button_create_new.setMinimumWidth(120)
+        vbox_create_new.addWidget(button_create_new, alignment=Qt.AlignLeft)
+        widget_create_new = QWidget()
+        widget_create_new.setLayout(vbox_create_new)
+        vbox_create_new.setContentsMargins(0, 0, 0, 0)
+        self.layout().addWidget(widget_create_new)
+        self.layout().addStretch(1)
+
+        temp_storage = None  # type: Optional[WalletStorage]
+        wallet_folder = os.path.dirname(path)
+
+        def on_choose():
+            path, __ = QFileDialog.getOpenFileName(self, "Select your wallet file", wallet_folder)
+            if path:
+                self.name_e.setText(path)
+
+        def on_filename(filename):
+            # FIXME? "filename" might contain ".." (etc) and hence sketchy path traversals are possible
+            nonlocal temp_storage
+            temp_storage = None
+            msg = None
+            self.wallet_exists = False
+            self.wallet_is_open = False
+            self.wallet_needs_hw_unlock = False
+            if filename:
+                path = os.path.join(wallet_folder, filename)
+                wallet_from_memory = self.wizard._daemon.get_wallet(path)
+                try:
+                    if wallet_from_memory:
+                        temp_storage = wallet_from_memory.storage  # type: Optional[WalletStorage]
+                        self.wallet_is_open = True
+                    else:
+                        temp_storage = WalletStorage(path)
+                    self.wallet_exists = temp_storage.file_exists()
+                except (StorageReadWriteError, WalletFileException) as e:
+                    msg = _('Cannot read file') + f'\n{repr(e)}'
+                except Exception as e:
+                    self.logger.exception('')
```

Comment:
`self.logger` looks undefined. Probably in some other instances too. In some cases it's a `_logger` vs `logger` issue, but here neither is defined I think.
Maybe we should standardise on one, between `_logger` and `logger`.

Reply:
For the `WizardComponent` descendants I used Logger.
For `QENewWalletWizard` that approach still has an issue.